File: client/lib/models/ride.dart
class Ride {
  final String id;
  final String source;
  final String destination;
  final String date;
  final String departureTime;
  final String arrivalTime;
  final String driver;
  final String vehicleName;
  final String vehicleColor;
  final double fare;

  Ride({
    required this.id,
    required this.source,
    required this.destination,
    required this.date,
    required this.departureTime,
    required this.arrivalTime,
    required this.vehicleName,
    required this.vehicleColor,
    required this.driver,
    required this.fare,
  });
}

File: server/models/ride.js
const mongoose = require('mongoose');

const Schema = mongoose.Schema;

const rideSchema = new Schema({
    source: {
        type: String,
    },
    destination: {
        type: String,
    },
    date : {
        type: String,
    },
    departureTime: {
        type: String,

    },
    arrivalTime: {
        type: String,

    },

    vehicleName: {
        type: String,
    },

    vehicleColor: {
        type: String
    },
   
    driver: {
        type: String,
    },
    fare: {
        type: Number,
    }
    });

module.exports = mongoose.model('Ride', rideSchema);